aten/src/ATen/native/TensorAdvancedIndexing.cpp

```diff
@@ -385,8 +386,14 @@ Tensor & index_copy_(Tensor & self, int64_t dim, const Tensor & index, const Ten
                    source.dim(), "), destination dimensionality (", self.dim(), ")");
   }
 
+  // Should this be TORCH_CHECK?
   TORCH_CHECK_INDEX(index.scalar_type() == ScalarType::Long, "index_copy_(): Expected LongTensor for index");
 
+  // Added in accordance with index_fill_
+  if (!self.is_complex() && source.is_complex()) {
+    TORCH_CHECK(false, "index_copy_(): Converting complex source Tensor to non-complex type is forbidden");
```

Comment:
"Tensor" -> "tensor"

"is forbidden" -> "is not supported"

Reply:
This doesn't seem sufficient. Shouldn't the src and dst tensors be checked for having the SAME dtype?

Another question, is it ever validated that all 3 tensors are on the same device?